File: src/main/java/com/pocketempire/config/FactionConfig.java
package com.pocketempire.config;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class FactionConfig {
    private String id;
    private String name;
    private String theme;
    private String icon;
    private String color;
    private String description;

    /*
    private int militaryState;
    private int economyState;
    private int scienceStat;
    */

    private int atkBonus;
    private int movementBonus;
    private int entrenchBonus;
    private double effectChanceBonus;
    private int forestDefBonus;

    private int lightUnitCostReduction;
    private int catapultCostReduction;
    private double researchMultiplier;
    private int wallCostReductionPercent;

    private int startingReputation;
    private int startingGold;
    private int borderContactRepDrain;
    private int betrayalThreshold;

    private List<String> preferredUnits;
    private String aiPersonality;
}

File: src/main/kotlin/com/pocketempire/config/FactionConfig.kt
package com.pocketempire.config

data class FactionConfig(
    var id: String = "",
    var name: String = "",
    var theme: String = "",
    var icon: String = "",
    var color: String = "",
    var description: String = "",

    var atkBonus: Int = 0,
    var movementBonus: Int = 0,
    var entrenchBonus: Int = 0,
    var effectChanceBonus: Double = 0.0,
    var forestDefBonus: Int = 0,
    var heavyHpBonus: Int = 0,
    var sightBonus: Int = 0,

    var lightUnitCostReduction: Int = 0,
    var catapultCostReduction: Int = 0,
    var mageCostReduction: Int = 0,
    var researchMultiplier: Double = 1.0,
    var wallCostReductionPercent: Int = 0,

    var startingReputation: Int = 0,
    var startingGold: Int = 0,
    var borderContactRepDrain: Int = 0,
    var betrayalThreshold: Int = 0,

    var warAtkBonusPerWar: Int = 0,
    var goldDividendPercent: Double = 0.0,

    var preferredUnits: List<String> = emptyList(),
    var aiPersonality: String = ""
)